src/utils.js

```diff
@@ -407,15 +408,10 @@ module.exports = (function() {
       var flickrURL = url + "?" + queryString + signature;
 
       var req = request.post(flickrURL, function(error, response, body) {
-        // format:json does not actually work, so we need to grab the photo ID from the response XML:
-        // <?xml version="1.0" encoding="utf-8" ?>\n<rsp stat="ok">\n<photoid>.........</photoid>\n</rsp>\n
-        var data;
-        if(!body) {
-          error = error || "No body found in response";
-        } else if (body.indexOf('rsp stat="ok"')>-1) {
-          data = parseInt(body.split("<photoid>")[1].split("</photoid>")[0], 10);
-        }
-        callback(error, data);
+	      if(!body) return callback("No body found in response");
+        var data = JSON.parse(parser.toJson(body));
+	      if (data.rsp.stat === "fail") return callback(data.rsp.err,null);
+        callback(null, data.rsp.photoid);
```

Comment:
XML is technically string data, so it's a good idea to make sure we still have `parseInt(data.rsp.photoid)` here.

Reply:
didn't know if always is an integer photo.id, but if that's the case, then would be a good idea